src/Serialization/ParseNodeFromStringTrait.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Microsoft\Kiota\Abstractions\Serialization;
+
+
+use DateInterval;
+use Exception;
+
+/**
+ * This trait contains utility functions for parsing strings to different types.
+ */
+trait ParseNodeFromStringTrait
+{
+    /**
+     * @throws Exception
+     */
+    public function getDateIntervalFromString(string $value): DateInterval
```

Comment:
shouldn't this be a constructor overload instead? (if possible in PHP) or a static method on the class?

Reply:
The class doesn't support using a static method call to set the `$invert` property. PHP doesn't support method overloading(including constructor).

Reply:
The method name sounds more like `parseDateIntervalFromString`  that `get` - get gives the notion of class properties

Reply:
+1 on the naming comment from Shem.
but surely when you parse the string, you'll create a new instance of the class, why can't the invert property be set then?

Reply:
The class doesn't handle the negative sign when parsing the string.

Reply:
ah, I think I get it now, we don't own this class, it comes from PHP directly.
forget my comment about making it a static method within the class then.
I'm guessing the trait semantics are more idiomatic in PHP than using a static helper class?

Reply:
Yes, the class is a PHP internal class.
Yes, traits can be used in place for static helper classes.